File: sdk-ios/Tune/Tune/TuneLocation.m
//
//  TuneLocation.m
//  MobileAppTracker
//
//  Created by Harshal Ogale on 8/3/15.
//  Copyright (c) 2015 TUNE. All rights reserved.
//

#import "TuneLocation+Internal.h"

@implementation TuneLocation

- (id)copyWithZone:(NSZone *)zone {
    TuneLocation *location = [[[self class] allocWithZone:zone] init];
    
    location.longitude = [self.longitude copyWithZone:zone];
    location.latitude = [self.latitude copyWithZone:zone];
    location.altitude = [self.altitude copyWithZone:zone];
    
    location.horizontalAccuracy = [self.horizontalAccuracy copyWithZone:zone];
    location.verticalAccuracy = [self.verticalAccuracy copyWithZone:zone];
    location.timestamp = [self.timestamp copyWithZone:zone];
    
    return location;
}

#pragma mark - NSCoding

- (id)initWithCoder:(NSCoder *)aDecoder {
    self = [super init];
    if (!self) {
        return nil;
    }
    
    self.longitude = [aDecoder decodeObjectForKey:@"longitude"];
    self.latitude = [aDecoder decodeObjectForKey:@"latitude"];
    self.altitude = [aDecoder decodeObjectForKey:@"altitude"];
    self.horizontalAccuracy = [aDecoder decodeObjectForKey:@"horizontalAccuracy"];
    self.verticalAccuracy = [aDecoder decodeObjectForKey:@"verticalAccuracy"];
    self.timestamp = [aDecoder decodeObjectForKey:@"timestamp"];
    
    return self;
}

- (void)encodeWithCoder:(NSCoder *)aCoder {
    [aCoder encodeObject:self.longitude forKey:@"longitude"];
    [aCoder encodeObject:self.latitude forKey:@"latitude"];
    [aCoder encodeObject:self.altitude forKey:@"altitude"];
    [aCoder encodeObject:self.horizontalAccuracy forKey:@"horizontalAccuracy"];
    [aCoder encodeObject:self.verticalAccuracy forKey:@"verticalAccuracy"];
    [aCoder encodeObject:self.timestamp forKey:@"timestamp"];
}

#pragma mark -

- (NSString *)debugDescription {
    return [NSString stringWithFormat:@"<%@: %p> %@, %@, %@, %@, %@, %@", [self class], (void *)self, self.latitude, self.longitude, self.altitude, self.horizontalAccuracy, self.verticalAccuracy, self.timestamp];
}

@end

File: sdk-android/TuneMarketingConsoleSDK/src/main/java/com/tune/TuneLocation.java
package com.tune;

import android.location.Location;

/**
 * Created by charlesgilliam on 1/22/16.
 */
public class TuneLocation {
    double altitude;
    double longitude;
    double latitude;

    public TuneLocation(Location location) {
        this.altitude = location.getAltitude();
        this.longitude = location.getLongitude();
        this.latitude = location.getLatitude();
    }

    public TuneLocation(double longitude, double latitude) {
        this.longitude = longitude;
        this.latitude = latitude;
    }

    public double getAltitude() {
        return altitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public TuneLocation setLongitude(double longitude) {
        this.longitude = longitude;
        return this;
    }

    public double getLatitude() {
        return latitude;
    }

    public TuneLocation setLatitude(double latitude) {
        this.latitude = latitude;
        return this;
    }
}
